org.eclipse.m2e.pde.ui/src/org/eclipse/m2e/pde/ui/target/editor/internal/part/AbstractDependencyEditorPart.java

```diff
@@ -0,0 +1,35 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Patrick Ziegler and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *      Patrick Ziegler - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.m2e.pde.ui.target.editor.internal.part;
+
+import org.eclipse.m2e.pde.ui.target.editor.internal.model.TargetDependencyModel;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Layout;
+
+public abstract class AbstractDependencyEditorPart {
```

Comment:
This super-class is only used to store fields that both sub-classes have. AFAICT it is never used as type and its only method is only used once. Therefore I'm not sure this is really necessary.

Reply:
That one was bugging me as well. I initially created the class, because I need to expose the parent composite, in order to make sure that the table viewer is only as large as the height of all buttons combined.
However, I achieved the same effect by moving this information to the model, meaning this class can go for good. :)

Reply:
Sounds good. 👍🏽